services/src/main/java/org/keycloak/protocol/oidc/utils/RedirectUtils.java

```diff
@@ -195,4 +196,27 @@ private static String getSingleValidRedirectUri(Collection<String> validRedirect
         }
         return validRedirect;
     }
+
+    public static String getValidRedirectUriWithRootUrl(ClientModel client) {
+        return (client != null) ? getValidRedirectUriWithRootUrl(client.getRootUrl(), client.getRedirectUris()) : null;
+    }
+
+    public static String getValidRedirectUriWithRootUrl(String rootUrl, Collection<String> redirectUris) {
```

Comment:
I see there is existing method `RedirectUtils.resolveValidRedirects(KeycloakSession session, String rootUrl, Set<String> validRedirects) ` , which seems to already quite similar thing you introduced here? Will it be possible to use the existing method as much as possible to avoid some duplicated code logic?

Reply:
Thanks. You're right.